asio/include/asio/detail/impl/throw_error.ipp

```diff
@@ -34,7 +34,7 @@ void do_throw_error(const asio::error_code& err)
 void do_throw_error(const asio::error_code& err, const char* location)
 {
   // boostify: non-boost code starts here
-#if defined(ASIO_MSVC) && defined(ASIO_HAS_STD_SYSTEM_ERROR)
+#if defined(ASIO_MSVC) && defined(ASIO_HAS_STD_SYSTEM_ERROR) && (!defined _MSC_VER || _MSC_VER < 1928)
```

Comment:
You don't need to check if `_MSC_VER` exists, because `ASIO_MSVC` pretty much guarantees that.

Reply:
thanks :)